temporal-test-server/src/main/java/io/temporal/internal/testservice/StateMachines.java

```diff
@@ -1222,6 +1221,16 @@ private static void bufferQuery(
     data.queryBuffer.put(query.getKey(), query);
   }
 
+  private static void bufferUpdate(
+      RequestContext ctx, WorkflowTaskData data, UpdateWorkflowExecution update, long notUsed) {
+    data.updateRequestBuffer.put(update.getId(), update);
```

Comment:
Need to validate one not already present for this ID here or on "add"?

Reply:
If you check there, good enough. I admit not following the flow all the way through.

Reply:
I added a check anyway, can't hurt